lib/edition_diff.rb

```diff
@@ -12,6 +12,14 @@ def field_diff
       diff.map(&:first)
   end
 
+  def full_diff
+    if previous_edition.present?
+      HashDiff.diff(current_edition.attributes, previous_edition.attributes)
+    else
+      HashDiff.diff(current_edition.attributes, previous_edition)
+    end
```

Comment:
To safe a conditional you could use the [safe navigation operator](http://mitrev.net/ruby/2015/11/13/the-operator-in-ruby/) 

I think we might need to use something more than attributes though so that we include things like links - not sure what else is part of an edition that isn't associated.